cargo-miri/src/util.rs

```diff
@@ -252,3 +252,20 @@ pub fn debug_cmd(prefix: &str, verbose: usize, cmd: &Command) {
     }
     eprintln!("{prefix} running command: {cmd:?}");
 }
+
+/// Checks if Cargo's build-std setting is enabled for one or more crates.
+pub fn is_build_std_set() -> bool {
+    let cmd = cargo()
```

Comment:
You need to pass `cargo_extra_flags()` as arguments to ensure that if the user sets `--cfg unstable.build-std` that will be respected. Also what about `-Zbuild-std`? I don't think I understand how these flags even interact.

It might be worth having a little helper function that invokes `cargo config` to get the value of an arbitrary key.